Source/Scene/ShadowMap.js

```diff
@@ -307,110 +455,292 @@ define([
+        var cameraNear = sceneCamera.frustum.near;
+        var cameraFar = sceneCamera.frustum.far;
+        var numberOfCascades = shadowMap._numberOfCascades;
+
+        // Split cascades. Use a mix of linear and log splits.
+        var lambda = 0.5;
+        var range = cameraFar - cameraNear;
+        var ratio = cameraFar / cameraNear;
+
+        var splitsNear = scratchSplitNear;
+        var splitsFar = scratchSplitFar;
+        splitsNear[0] = cameraNear;
+        splitsFar[numberOfCascades - 1] = cameraFar;
+
+        for (var i = 0; i < numberOfCascades - 1; ++i) {
+            var p = (i + 1) / numberOfCascades;
+            var logScale = cameraNear * Math.pow(ratio, p);
+            var uniformScale = cameraNear + range * p;
+            var split = CesiumMath.lerp(uniformScale, logScale, lambda);
+            splitsFar[i] = split;
+            splitsNear[i + 1] = split;
+        }
+        Cartesian4.unpack(splitsNear, 0, shadowMap._cascadeSplits[0]);
+        Cartesian4.unpack(splitsFar, 0, shadowMap._cascadeSplits[1]);
+
+        var left = shadowMapCamera.frustum.left;
+        var right = shadowMapCamera.frustum.right;
+        var bottom = shadowMapCamera.frustum.bottom;
+        var top = shadowMapCamera.frustum.top;
+        var near = shadowMapCamera.frustum.near;
+        var far = shadowMapCamera.frustum.far;
+
+        var cascadeSubFrustum = sceneCamera.frustum.clone(scratchFrustum);
+
+        for (var j = 0; j < numberOfCascades; ++j) {
+            // Find the bounding box of the camera sub-frustum in shadow map texture space
+            cascadeSubFrustum.near = splitsNear[j];
+            cascadeSubFrustum.far = splitsFar[j];
+            var viewProjection = Matrix4.multiply(cascadeSubFrustum.projectionMatrix, sceneCamera.viewMatrix, scratchMatrix);
+            var inverseViewProjection = Matrix4.inverse(viewProjection, scratchMatrix);
+            var cascadeMatrix = Matrix4.multiply(shadowMapCamera.getViewProjection(), inverseViewProjection, scratchMatrix);
+
+            // Project each corner from camera NDC space to shadow map texture space. Min and max will be from 0 to 1.
+            var min = Cartesian3.fromElements(Number.MAX_VALUE, Number.MAX_VALUE, Number.MAX_VALUE, scratchMin);
+            var max = Cartesian3.fromElements(-Number.MAX_VALUE, -Number.MAX_VALUE, -Number.MAX_VALUE, scratchMax);
+
+            for (var k = 0; k < 8; ++k) {
+                var corner = Cartesian4.clone(frustumCornersNDC[k], scratchFrustumCorners[k]);
+                Matrix4.multiplyByVector(cascadeMatrix, corner, corner);
+                Cartesian3.divideByScalar(corner, corner.w, corner); // Handle the perspective divide
+                Cartesian3.minimumByComponent(corner, min, min);
+                Cartesian3.maximumByComponent(corner, max, max);
+            }
+
+            // Limit min to 0.0, sometimes precision errors cause it to go slightly negative.
+            min.x = Math.max(min.x, 0.0);
```

Comment:
I could clamp, but it's not as serious. The negative number was messing up the frustum sides, whereas > 1.0 would just extend the sides slightly.